lisa/util/process.py

```diff
@@ -230,6 +230,17 @@ def wait_result(
         timer = create_timer()
         is_timeout = False
 
+        if (
+            self._sudo
+            and isinstance(self._shell, SshShell)
+            and self._shell._is_sudo_required_password
+        ):
+            # Add a 0.5s delay to make sure the channel is ready to receive
+            time.sleep(0.5)
+            if self.is_running():
+                assert self._process
+                self._process.stdin_write(f"{self._shell._connection_info.password}\n")
```

Comment:
I think in some case, it doesn't need to input password for every commands. I see some system ask for password every 15 minutes. So it needs to check if the output buffer contains the password prompt, and then input it.

Reply:
Yeah, put the check here is better. So we don't need the check in suport_sodo, right?

Reply:
> Yeah, put the check here is better. So we don't need the check in suport_sodo, right?

Not sure I fully understand. check password required or not happens only once for a shell or node, so it's hard to be done in process level, it's ok to put it in node level, but shell maybe ok as well. When the sudo password is required, it has two situations, 1) need it every time, 2) need it every x minutes. It needs to be detected in process level, so the password can be hacked into wait_result.

for the two situations, it needs to check if password is needed by the related prompt.

Reply:
Changed to check in suport_sudo. From the images I have seen, they all have this behavior that after inputting password successfully, it no needs to input password again if in the same session. If in a new session, it still needs to input password. In Lisav3, every command is a new session, so needs input password every time.

As we discussed, I changed to "echo OK" twice, then I can get all kinds of password prompts.